### PR TITLE
Fixes #3059 #2996 - Improved Blog Page

### LINKED ISSUE #3059: Next and Previous buttons of the blog not displayed.
**Actual Behaviour**

The Next and previous buttons (implemented in the index.js) of the blog aren't displayed as expected and because of this the users are unable to navigate/view older blog posts as expected.  

**Expected Behaviour**

Users should be able to view all the blog posts that have been published.

**Screenshots of the issue**

(Notice no Previous or Next Fab buttons in the bottom)
NoButtons

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by AuraOfDivinity**
Fixes #3059 #2996 

**Note:** _This PR is a follow up and implements the requested changes at https://github.com/fossasia/susi.ai/pull/3023 Unfortunately I had a change of PC's and I decided to start off with a new remote and a local repository, hence this new PR. Apologies for any inconvenience caused._

Changes:
1) Added in a search bar to search for particular blog posts.
2) Blog posts are now handled as separate components.
3) Blog posts initially have a small part of the post which could be expanded to view the entire post using the Read More button.
4) Changed the size and styling of the Facebook / Twitter sharing buttons to make them look more prominent.
5) Removed the old pagination and re implemented it from the ground up as it was not working properly.

Demo Link: https://pr-3076-fossasia-susi-web-chat.surge.sh

Screenshots of the change:

ImprovedBlog

### CODE REVIEW

**Comment by AuraOfDivinity** [comment]
@akshatnitd @plxity I have made the requested changes at https://github.com/fossasia/susi.ai/pull/3023 in this PR. Please do take a look and let me know if any other changes are required. Thanks!

File: src/components/About/Blog/BlogPost.js [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+import styled, { css } from 'styled-components';
+import Fade from '@material-ui/core/Fade';
+import './Blog.css';
+import dateFormat from 'dateformat';
+import htmlToText from 'html-to-text';
+import renderHTML from 'react-render-html';
+import susi from '../../../images/susi-logo.svg';
+import PropTypes from 'prop-types';
+
+const allCategories = [
```
Comment (akshatnitd) [inline src/components/About/Blog/BlogPost.js]: Move this to constants folder.

File: src/components/About/Blog/BlogPost.js [hunk 2]
```diff
@@ -0,0 +1,339 @@
...
+  'Event Management',
+  'loklak',
+  'Meilix',
+  'Open Event',
+  'Phimpme',
+  'Pocket Science Lab',
+  'yaydoc',
+];
+
+const arrDiff = (a1, a2) => {
```
Comment (akshatnitd) [inline src/components/About/Blog/BlogPost.js]: Are you trying to intersect 2 arrays here?

Check out lodash operations and use directly.

File: src/components/About/Blog/BlogPost.js [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+  width: 100%;
+  display: flex;
+  padding: 0.625rem 0 1.25rem 0.625rem;
+`;
+
+class BlogPost extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      expandedElements: [],
```
Comment (akshatnitd) [inline src/components/About/Blog/BlogPost.js]: Why is this required as array? Bool variable named `isExpanded` should be fine IMO.

File: src/components/About/Blog/BlogPost.js [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+    };
+    this.expandedElements = [];
+    this.post = this.props.posts;
```
Comment (akshatnitd) [inline src/components/About/Blog/BlogPost.js]: Also, why are these 2 variables required.

File: src/components/About/Blog/BlogPost.js [hunk 5]
```diff
@@ -0,0 +1,339 @@
...
+      let k = 0;
+      for (k = 0; k < allCategories.length; k++) {
+        if (cat === allCategories[k]) {
+          category.push(cat);
+        }
+      }
+    });
+
+    const tags = arrDiff(category, this.post.categories);
+    const fCategory = category
```
Comment (akshatnitd) [inline src/components/About/Blog/BlogPost.js]: Please refactor these logic and move to separate function.

**Comment by AuraOfDivinity** [comment]
@akshatnitd  I have made the requested changes. Kindly have a look and let me know if any other changes are required!

File: src/components/About/Blog/BlogPost.js [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+class BlogPost extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      isExpanded: false,
+    };
+    this.post = this.props.posts;
+  }
+
+  handleClickExpand = () => {
```
Comment (akshatnitd) [inline src/components/About/Blog/BlogPost.js]: Instead of 2 functions here. Make a `toggleCardExpansion` function.

File: src/components/About/Blog/BlogPost.js [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+    });
+  };
+
+  handleClickShrink = () => {
+    this.setState({
+      isExpanded: false,
+    });
+  };
+
+  getfCategories = category => {
```
Comment (akshatnitd) [inline src/components/About/Blog/BlogPost.js]: What is `fCategories, fCategory, getfTags, fTag`? Use proper variable naming.

File: src/components/About/Blog/BlogPost.js [hunk 8]
```diff
@@ -0,0 +1,325 @@
...
+          {tag}
+        </BlogFooterLink>
+      ))
+      .reduce((prev, curr) => [prev, ', ', curr]);
+
+    return ftag;
+  };
+
+  getResultText = content => {
+    var dummyNode = document.createElement('div'),
```
Comment (akshatnitd) [inline src/components/About/Blog/BlogPost.js]: Why do you want to use `document. .....` APIs?

**Comment by AuraOfDivinity** [comment]
@akshatnitd  Will definitely do this week! My exams will be over by the end of this week!

**Comment by akshatnitd** [comment]
@AuraOfDivinity Please close the PR, if you are not working on it. Please make way for the active contributors.

**Comment by akshatnitd** [comment]
Closing this PR as most of the requirement enhancements are already done.
